wagtail/wagtailsearch/backends/elasticsearch.py

```diff
@@ -509,9 +510,18 @@ def add_model(self, model):
         mapping = self.mapping_class(model)
 
         # Put mapping
-        self.es.indices.put_mapping(
-            index=self.name, doc_type=mapping.get_document_type(), body=mapping.get_mapping()
-        )
+        try:
+            self.es.indices.put_mapping(
+                index=self.name, doc_type=mapping.get_document_type(), body=mapping.get_mapping()
+            )
+        except RequestError as e:
+            if 'No handler for type [attachment] declared' in str(e.info):
+                raise AssertionError(
```

Comment:
I don't agree with swapping the ``RequestError`` for a plain ``AssertionError`` here.

Perhaps we could raise a ``RuntimeWarning`` with the hint instead?

I think this belongs in the docs as well.